src/plot_api/plot_api.js

```diff
@@ -2074,7 +2072,7 @@ function _relayout(gd, aobj) {
                 if(manageArrays.isAddVal(vi)) {
                     undoit[ai] = null;
                 } else if(manageArrays.isRemoveVal(vi)) {
-                    undoit[ai] = obji;
+                    undoit[ai] = (nestedProperty(layout, arrayStr).get() || [])[i];
```

Comment:
We might not have anyone who cares about undo/redo anymore, but we have it so here's a fix. We were missing the `.get()` the way this was written before.

Reply:
We have a tentative plan to :hocho: undo/redo in v2 https://github.com/plotly/plotly.js/issues/420#issuecomment-322873840, with the rationale:

> we used these in our old workspace, but this is really not the right level to be managing this issue, as plots may be coupled to other application state (ie changes in data arrays) and using our queue would muck up that correspondence.

so unless we reverse that conclusion let's not build anything else using it.